formats/notion/convert-to-md.ts

```diff
@@ -19,226 +23,226 @@ export function convertNotesToMd({
 	attachmentFolderPath: string;
 }) {
 	for (let fileInfo of Object.values(idsToFileInfo)) {
-		fileInfo.body = convertInlineDatabasesToObsidian(fileInfo.body, {
-			idsToFileInfo,
-			pathsToAttachmentInfo,
-			attachmentFolderPath,
-		});
+		// fileInfo.body = convertInlineDatabasesToObsidian(fileInfo.body, {
+		// 	idsToFileInfo,
+		// 	pathsToAttachmentInfo,
+		// 	attachmentFolderPath,
+		// });
 
-		fileInfo.body = convertLinksToObsidian(fileInfo.body, {
+		convertLinksToObsidian(fileInfo.notionLinks, {
 			idsToFileInfo,
 			pathsToAttachmentInfo,
 			attachmentFolderPath,
 			fileInfo,
 		});
 
-		replaceNestedFormatting(fileInfo.body);
-		replaceNewlinesInFormatting(fileInfo.body);
-		encodeNewlines(fileInfo.body);
-		stripLinkImages(fileInfo.body);
+		replaceNestedTags(fileInfo.body, 'strong');
+		replaceNestedTags(fileInfo.body, 'em');
+		stripLinkFormatting(fileInfo.body);
+		encodeNewlinesToBr(fileInfo.body);
+		splitBrsInFormatting(fileInfo.body, 'strong');
+		splitBrsInFormatting(fileInfo.body, 'em');
 		fixNotionDates(fileInfo.body);
 		fixNotionLists(fileInfo.body);
 		replaceTableOfContents(fileInfo.body);
-		fileInfo.body = htmlToMarkdown(fileInfo.body);
-		fileInfo.body = escapeHashtags(fileInfo.body);
+
+		fileInfo.markdownBody = htmlToMarkdown(fileInfo.body.innerHTML);
+		fileInfo.markdownBody = escapeHashtags(fileInfo.markdownBody);
 
 		if (fileInfo.properties) {
-			fileInfo.yamlProperties = fileInfo.properties
-				.filter((property) => property.content)
-				.map((property) =>
-					convertPropertyToYAML(property, {
-						fileInfo,
-						idsToFileInfo,
-						pathsToAttachmentInfo,
-						attachmentFolderPath,
-					})
-				);
+			fileInfo.yamlProperties = fileInfo.properties.map((property) =>
+				convertPropertyToYAML(property, {
+					fileInfo,
+					idsToFileInfo,
+					pathsToAttachmentInfo,
+					attachmentFolderPath,
+				})
+			);
 		}
 	}
 }
 
-const replaceNestedFormatting = (body: string) => {
-	body = body
-		.replace(/<strong>(<strong>)+/g, '<strong>')
-		.replace(/<\/strong>(<\/strong>)+/g, '</strong>')
-		.replace(/<em>(<em>)+/g, '<em>')
-		.replace(/<\/em>(<\/em>)+/g, '</em>');
+const replaceNestedTags = (body: HTMLElement, tag: 'strong' | 'em') => {
+	body.querySelectorAll(tag).forEach((el) => {
+		const nestedEls = el.querySelectorAll(tag);
+		if (nestedEls.length === 0) return;
+		const textContent = el.textContent;
```

Comment:
How about hoisting the children (moving all children elements of the inner element up) instead of flattening it to `textContent`? It would help prevent destroying other formatting or non-text element like images and links.

Reply:
Seems much better. I'll do that.